examples/vision/knowledgedistillation.py

```diff
@@ -0,0 +1,224 @@
+                tf.nn.softmax(s_pred / self.T, axis=1),
+            )
+            loss = self.alpha * s_loss + (1 - self.alpha) * d_loss
+
+        # Compute gradients
+        trainable_vars = self.student.trainable_variables
+        gradients = tape.gradient(loss, trainable_vars)
+
+        # Update weights
+        self.optimizer.apply_gradients(zip(gradients, trainable_vars))
+
+        # Update the metrics configured in `compile()`.
+        self.compiled_metrics.update_state(y, s_pred)
+
+        # Return a dict of performance
+        return {
+            **{"s_loss": s_loss, "d_loss": d_loss},
+            **{m.name: m.result() for m in self.metrics},
+        }
+
+    def test_step(self, data):
+        # Unpack the data
+        x, y = data
+
+        # Compute predictions
+        y_pred = self.student(x, training=False)
+
+        # Calculate the loss
+        s_loss = self.s_loss_fn(y, y_pred)
+
+        # Update the metrics.
+        self.compiled_metrics.update_state(y, y_pred)
+
+        # Return a dict of performance
+        return {**{"s_loss": s_loss}, **{m.name: m.result() for m in self.metrics}}
+
+
+"""
+## Create student and teacher models
+Initialy, we create a teacher model and a smaller student model. Both models are
+convolutional neural networks and created using `Sequential()`, but could be any model
+created with the Keras API.
+"""
+
+# Create the teacher
+teacher = keras.Sequential(
+    [
+        keras.Input(shape=(28, 28, 1)),
+        layers.Conv2D(256, (3, 3), strides=(2, 2), padding="same"),
+        layers.LeakyReLU(alpha=0.2),
+        layers.MaxPooling2D(pool_size=(2, 2), strides=(1, 1), padding="same"),
+        layers.Conv2D(512, (3, 3), strides=(2, 2), padding="same"),
+        layers.Flatten(),
+        layers.Dense(10),
+    ],
+    name="teacher",
+)
+
+# Create the student
+student = keras.Sequential(
+    [
+        keras.Input(shape=(28, 28, 1)),
+        layers.Conv2D(16, (3, 3), strides=(2, 2), padding="same"),
+        layers.LeakyReLU(alpha=0.2),
+        layers.MaxPooling2D(pool_size=(2, 2), strides=(1, 1), padding="same"),
+        layers.Conv2D(32, (3, 3), strides=(2, 2), padding="same"),
+        layers.Flatten(),
+        layers.Dense(10),
+    ],
+    name="student",
+)
+
+"""
+## Prepare dataset
+The dataset used for training the teacher and distilling the teacher is
+[MNIST](https://keras.io/api/datasets/mnist/). Both the student and teacher are trained
+on the training set and evaluated on the test set.
+"""
+
+# Prepare the train and test dataset.
+batch_size = 64
+(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()
+
+# Normalize data
+x_train = x_train.astype("float32") / 255.0
+x_train = np.reshape(x_train, (-1, 28, 28, 1))
+
+x_test = x_test.astype("float32") / 255.0
+x_test = np.reshape(x_test, (-1, 28, 28, 1))
+
+# Construct datasets
+train_dataset = tf.data.Dataset.from_tensor_slices((x_train, y_train))
```

Comment:
Since your data is already NumPy arrays, you don't need to create datasets, just pass the arrays directly to `fit`

Reply:
I use the dataset structure in order to reduce the amount of batches for training in the example in a concise manner. I am unsure of a pretty way to do this if we merely pass in the NumPy arrays.
It would be something like `.fit(x_train[:640], y_train[:640], batch_size=64, epochs=1)` compared to `.fit(train_dataset.take(10), epochs=1)`.